pkg/client/gcp/types.go

```diff
@@ -45,9 +47,32 @@ func (c *Client) ListKubernetesFirewallRulesForNetwork(ctx context.Context, proj
 	return firewalls, nil
 }
 
+// ListKubernetesRoutesForNetwork returns a list of all routes within the shoot network which have the namespace as prefix.
+func (c *Client) ListKubernetesRoutesForNetwork(ctx context.Context, project, network, namespace string) ([]string, error) {
+	var routes []string
+	if err := c.computeService.Routes.List(project).Pages(ctx, func(page *compute.RouteList) error {
+		for _, route := range page.Items {
```

Comment:
Have you checked for potential nil pointer?

Reply:
Do you mean for *compute.RouteList? There are at least the default routes included.